### PR TITLE
refactor: Upgrade influx-export to use V3

### PR DESCRIPTION
**Pull request by marcpfuller**
**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)

## Testing Instructions
Start edgex with no-secty
run `./app-service  -cp -o -dev`
verify service starts up

### CODE REVIEW

File: application-services/custom/influxdb-export/main.go [hunk 1]
```diff
@@ -99,7 +99,7 @@ func (c *ServiceConfig) Validate() error {
 		return errors.New("configuration missing value for MqttSecretConfig.AuthMode")
 	}
 
-	if len(strings.TrimSpace(c.MqttConfig.SecretPath)) == 0 {
+	if len(strings.TrimSpace(c.MqttConfig.SecretName)) == 0 {
 		return errors.New("configuration missing value for MqttSecretConfig.SecretPath")
```
Comment (jim-wang-intel) [inline application-services/custom/influxdb-export/main.go:103]:
```suggestion
	if len(strings.TrimSpace(c.MqttConfig.SecretName)) == 0 {
		return errors.New("configuration missing value for MqttSecretConfig.SecretName")
```

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]

**Review by jim-wang-intel — DISMISSED** [review]

File: application-services/custom/influxdb-export/res/configuration.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      path: "mqtt"
+      Secrets:
+        username: "edgex"
+        password: "password"
+        cacert: ""
+        clientcert: ""
+        clientkey: ""
+  Writable.Telemetry:
+    Interval: "30s"
+    PublishTopicPrefix : "edgex/telemetry" # /<service-name>/<metric-name> will be added to this Publish Topic prefix
```
Comment (lenny-intel) [inline application-services/custom/influxdb-export/res/configuration.yaml:14]: Remove. no longer exists
```suggestion
```

File: application-services/custom/influxdb-export/res/configuration.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+Trigger:
+  Type: "edgex-messagebus"
+  EdgexMessageBus:
+    Type: "redis"
+    SubscribeHost:
+      Host: "localhost"
+      Port: 6379
+      Protocol: "redis"
+      SubscribeTopics: "edgex/events/#"
```
Comment (lenny-intel) [inline application-services/custom/influxdb-export/res/configuration.yaml:29]: Remove and replace with comment like is done here: https://github.com/edgexfoundry/app-service-configurable/blob/main/res/mqtt-export/configuration.yaml#L84

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: application-services/custom/influxdb-export/res/configuration.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        clientkey: ""
+  Writable.Telemetry:
+    Interval: "30s"
+    PublishTopicPrefix : "edgex/telemetry" # /<service-name>/<metric-name> will be added to this Publish Topic prefix
+
+Service:
+  Host: "localhost"
+  Port: 59780 # Adjust if running multiple examples at the same time to avoid duplicate port conflicts
+  StartupMsg: "App service export to InfluxDB has Started"
+
```
Comment (lenny-intel) [inline application-services/custom/influxdb-export/res/configuration.yaml:20]: Missing MessageBus section for client ID override

```
MessageBus:
  Disabled: false  # Set to true if not using edgex-messagebus Trigger below and Metrics are enabled
  Optional:
    ClientId: "app-influxdb-export"
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: application-services/custom/influxdb-export/res/configuration.yaml (hunk 2 above)
Comment (marcpfuller) [reply application-services/custom/influxdb-export/res/configuration.yaml]: done

File: application-services/custom/influxdb-export/res/configuration.yaml (hunk 3 above)
Comment (marcpfuller) [reply application-services/custom/influxdb-export/res/configuration.yaml]: done

File: application-services/custom/influxdb-export/res/configuration.yaml (hunk 4 above)
Comment (marcpfuller) [reply application-services/custom/influxdb-export/res/configuration.yaml:19]: done

**Review by jim-wang-intel — APPROVED** [review]

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**